src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -74,5 +76,121 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
 #endif
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+    CHIP_ERROR err = ConfigurationMgr().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrReturnError(count <= UINT16_MAX, CHIP_ERROR_INVALID_INTEGER_VALUE);
+        rebootCount = static_cast<uint16_t>(count);
+    }
+
+    return err;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(uint8_t & bootReason)
+{
+    uint32_t reason = 0;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (hwinfo_get_reset_cause(&reason) != 0)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    if (reason & (RESET_POR | RESET_PIN))
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_POWER_ON_REBOOT;
+    }
+    else if (reason & RESET_WATCHDOG)
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_SOFTWARE_WATCHDOG_RESET;
```

Comment:
Isn't it from the hardware watchdog?

Reply:
I'm not sure, taking look at the nrf implementation: https://github.com/nrfconnect/sdk-zephyr/blob/main/drivers/hwinfo/hwinfo_nrf.c#L46 and then to the HAL I found it basically may consider both watchdogs: https://github.com/nrfconnect/sdk-hal_nordic/blob/main/nrfx/helpers/nrfx_reset_reason.h#L65 The other DOG1 mask describes only application watchdog: https://github.com/nrfconnect/sdk-hal_nordic/blob/main/nrfx/helpers/nrfx_reset_reason.h#L92, so I don't know if any option here will be correct.